app/celery/tasks.py

```diff
@@ -412,9 +412,9 @@ def save_emails(self, _service_id: Optional[str], signed_notifications: List[Sig
         # service is whatever it was set to last in the for loop above.
         # at this point in the code we have a list of notifications (saved_notifications)
         # which could be from multiple services
-        check_service_over_daily_message_limit(KEY_TYPE_NORMAL, service)
         research_mode = service.research_mode  # type: ignore
         for notification_obj in saved_notifications:
+            check_service_over_daily_message_limit(notification_obj.key_type, service)
```

Comment:
Should we have a default to `KEY_TYPE_NORMAL` just in case? Can we guarantee this will always be there?

Reply:
We are defaulting to `KEY_TYPE_NORMAL` here: https://github.com/cds-snc/notification-api/blob/bug/test-key-sending-blocked/app/celery/tasks.py#L366

So I think the `key_type` will always be there.

Reply:
Thanks @smcmurtry . I very much like the new SignedNotification and VerifiedNotification types btw.